app/routers/api/auth.py

```
@@ -244,3 +255,354 @@ def generate_api_token(
+        # Create new user
+        import secrets
+
+        random_password = secrets.token_urlsafe(32)
+        user = User(
+            username=username,
+            password_hash=hash_password(random_password),
+            is_active=True,
+            roles=final_roles,
+        )
+        db.add(user)
+        logger.info(f"OIDC: Created new user '{username}' with roles {final_roles}")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)